### PR TITLE
WIP: Migrate to ts

### LINKED ISSUE #1735: Convert the whole codebase to TypeScript and make it strict
- [x] Convert the remaining files to TypeScript: 
  https://github.com/sindresorhus/refined-github/tree/master/source/libs
  https://github.com/sindresorhus/refined-github/tree/master/source
- [ ] Enable strict mode: https://github.com/sindresorhus/refined-github/blob/11b927189bab258a226e251c1bd87504feee51f4/tsconfig.json#L10
- [ ] Enable these rules again and fix the lint errors: https://github.com/sindresorhus/refined-github/blob/e90b58c16f58f97a0758de05ae7fe01bf7f65687/package.json#L77-L78

### PR DESCRIPTION
**Pull request by jotatoledo**
https://twitter.com/sindresorhus/status/1091023787313684480

:100: 

Closes #1735

### CODE REVIEW

**Comment by bfred-it** [comment]
Thank you for starting this!

It will partially fix #1735

File: source/libs/indent-textarea.ts [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
-		el.setSelectionRange(
-			selectionStart + 1,
-			selectionEnd + linesCount
-		);
+		el.setSelectionRange(selectionStart + 1, selectionEnd + linesCount);
 	} else {
-		document.execCommand('insertText', false, '\t');
+		document.execCommand("insertText", false, "\t");
 	}
 }
```
[the remark this answers is not in the document]
Comment (busches) [reply source/libs/indent-textarea.ts]: They also fail in the lint step.

**Comment by sindresorhus** [comment]
I think it's best if you rename all the files you intend to work on from `.js` to `.ts` in the first commit, so it's easier to review and Git can follow the renames.

File: source/libs/indent-textarea.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jotatoledo) [reply source/libs/indent-textarea.ts]: Yup, prettier is executed by default on save. Ill fix up those changed :+1:

[the remark this answers is not in the document]
Comment (bfred-it) [reply source/libs/indent-textarea.ts]: It’s better to disable it

File: source/libs/utils.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				zipped.push(row[col]);
+				if (zipped.length === limit) {
+					return zipped;
+				}
+			}
+		}
+	}
+
+	return zipped;
+};
```
[the remark this answers is not in the document]
Comment (bfred-it) [reply source/libs/utils.ts]: 2 commits are fine. They'll all be squashed on merge anyway.

[the remark this answers is not in the document]
Comment (jotatoledo) [reply source/libs/utils.ts]: No worries, I found a way by simply using ammend to commit the changes after the renaming

**Comment by jotatoledo** [comment]
@sindresorhus @bfred-it cleaned up the commits. For some reason the icons.js appears deleted when the gitlog is displayed in github, locally its displayed as renamed.

Most likely the test/lint step needs to be tweaked, any tips would be appreciated as I have no experience with both xo and ava

File: source/libs/indent-textarea.ts [hunk 3]
```diff
@@ -1,28 +1,25 @@
-export default function (el) {
+export default function(el: HTMLTextAreaElement) {
```
[the remark this answers is not in the document]
Comment (jotatoledo) [reply source/libs/indent-textarea.ts]: addressed

File: source/libs/on-new-comments.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	observed.add(discussion);
+
+	// When new comments come in via AJAX
+	observeEl(discussion, run);
+
+	// When hidden comments are loaded by clicking "Load more..."
+	addListenersOnNewElements();
+};
+
+export default function(cb: Function) {
```
[the remark this answers is not in the document]
Comment (jotatoledo) [reply source/libs/on-new-comments.ts]: addressed

**Comment by jotatoledo** [comment]
Dropped in favor of #1750

**Comment by sindresorhus** [comment]
@jotatoledo Sorry that this ended up not being used. We appreciate your effort 🙌 We could use some help with reviewing #1750 if you're interested.

**Pull request closed without merging**